aips/aip-31.md

```diff
@@ -1,46 +1,124 @@
 ---
 aip: 31
-title: Allowlisting for delegation pool
-author: wintertoro, michelle-aptos, junkil-park, movekevin, alexfilip2
-discussions-to (*optional): <a url pointing to the official discussion thread>
-Status: Draft
+title: Allow listing for delegation pools
+author: junkil-park, movekevin, michelle-aptos, wintertoro, alexfilip2
+discussions-to: https://github.com/aptos-foundation/AIPs/issues/121
+Status: In Review
 last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
 type: Framework
 created: 5/3/2023
-updated (*optional): 02/07/2024
+updated: 4/16/2024
 ---
 
-# AIP-31 - Allowlisting for delegation pool
+# AIP-31 - Allowlisting for delegation pools
 
 ## Summary
 
-This AIP proposes to add a flag in `delegation_pool.move` to allow “permissioned” acceptance of stakers. 
+This AIP introduces the concept of allowlisting in delegation pools, allowing pool owners to control who can stake in their delegation pool. The feature includes mechanisms for adding or removing delegators from the allowlist and evicting delegators not on the allowlist.
 
-## Motivation
+## High-level overview
 
-Some delegation pools may require a KYC process before accepting delegated stake for compliance reasons. Since delegation pools are permissionless, this feature gives the pool owner more control over who the delegation pool can accept stake from. 
+Currently, all the delegation pools are permissionless, meaning that any staker can stake to any delegation pool. However, for various reasons such as compliance, some delegation pools may prefer to operate on a permissioned basis, restricting staking to only those from an allowlist of wallet addresses. This feature will enable pool owners to establish an allowlist, controlling who can stake in their delegation pool.
 
-## Rationale
+## Impact
 
-**Considerations:**
+This feature impacts several aspects:
+
+* Pool owners: Pool owners can set an allowlist for their delegation pool. They can add or remove addresses from the allowlist and evict delegators who are not on the allowlist.
+
+* Delegators: Delegators are unable to add stake or reactivate stake in a delegation pool if the pool is allowlisted and they are not included on the allowlist. They can also be evicted from the pool by the pool owner, resulting in their stake becoming unlocked. The evicted delegator can then withdraw their unlocked fund.
```

Comment:
if the pool has allow listing enabled, but they are not included on the allowlist.